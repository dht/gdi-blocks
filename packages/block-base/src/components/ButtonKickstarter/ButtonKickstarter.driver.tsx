import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ButtonKickstarter, ButtonKickstarterProps } from './ButtonKickstarter';
import { BaseComponentDriver } from 'testing-base';

export class ButtonKickstarterDriver extends BaseComponentDriver {
    private props: Partial<ButtonKickstarterProps> = {};

    constructor() {
        super('ButtonKickstarter');
    }

    when: any = {
        rendered: () => {
            render(<ButtonKickstarter {...(this.props as ButtonKickstarterProps)} />);
            return this;
        },
        clicked: () => {
            fireEvent.click(this.wrapper);
            return this;
        },
        snapshot: () => {
            return this.snapshot(<ButtonKickstarter {...(this.props as ButtonKickstarterProps)} />);
        },
    };

    given: any = {
        props: (props: Partial<ButtonKickstarterProps>) => {
            this.props = props;
            return this;
        },
    };

    get = {
        WrapperClassName: () => {
            return this.wrapper.className;
        },
        label: () => {
            return this.wrapper.innerHTML;
        },
    };
}
