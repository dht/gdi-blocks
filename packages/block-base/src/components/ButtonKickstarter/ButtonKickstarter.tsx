import React from 'react';
import { Wrapper } from './ButtonKickstarter.style';

export type ButtonKickstarterProps = {};

export function ButtonKickstarter(_props: ButtonKickstarterProps) {
    return (
        <Wrapper
            className='ButtonKickstarter-wrapper'
            data-testid='ButtonKickstarter-wrapper'
        >
            ButtonKickstarter
        </Wrapper>
    );
}

export default ButtonKickstarter;
