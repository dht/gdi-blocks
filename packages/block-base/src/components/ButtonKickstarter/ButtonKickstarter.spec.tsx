import { ButtonKickstarterDriver } from './ButtonKickstarter.driver';
import Chance from 'chance';

const chance = new Chance();

describe('ButtonKickstarter', () => {
    let driver: ButtonKickstarterDriver;

    beforeAll(() => {
        driver = new ButtonKickstarterDriver();
    });

    it('should render button', () => {
        const label = chance.word();

        const element = driver.given
            .props({
                title: label,
            })
            .when.rendered();

        const wrapperClassName = element.get.wrapperClassName();
        const innerText = element.get.label();

        expect(wrapperClassName).toContain('ButtonKickstarter-wrapper');
        expect(innerText).toBe(label);
    });

    it('should click button', () => {
        const callback = jest.fn();

        driver.given
            .props({
                onClick: callback,
            })
            .when.rendered()
            .when.clicked();

        expect(callback).toHaveBeenCalledTimes(1);
    });
});

describe('ButtonKickstarter snapshots', () => {
    let driver: ButtonKickstarterDriver;

    beforeAll(() => {
        driver = new ButtonKickstarterDriver();
    });

    it('should match snapshot', () => {
        expect(driver.when.snapshot()).toMatchSnapshot();
    });
});
