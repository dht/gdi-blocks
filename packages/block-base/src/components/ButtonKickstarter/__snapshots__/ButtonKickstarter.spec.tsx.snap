// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonKickstarter snapshots should match snapshot 1`] = `
<div
  className="sc-bcXHqe giHcUI ButtonKickstarter-wrapper"
  data-testid="ButtonKickstarter-wrapper"
>
  ButtonKickstarter
</div>
`;
